datasets/gooaq/gooaq.py

```diff
@@ -0,0 +1,122 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""GooAQ - Question-answers, collected from Google"""
+
+
+import json
+
+import datasets
+
+
+_CITATION = """\
+@article{gooaq2021,
+  title={GooAQ: Open Question Answering with Diverse Answer Types},
+  author={Khashabi, Daniel and Ng, Amos and Khot, Tushar and Sabharwal, Ashish and Hajishirzi, Hannaneh and Callison-Burch, Chris},
+  journal={arXiv preprint},
+  year={2021}
+}
+"""
+
+_DESCRIPTION = """\
+GooAQ is a large-scale dataset with a variety of answer types. This dataset contains over
+5 million questions and 3 million answers collected from Google. GooAQ questions are collected
+semi-automatically from the Google search engine using its autocomplete feature. This results in
+naturalistic questions of practical interest that are nonetheless short and expressed using simple
+language. GooAQ answers are mined from Google's responses to our collected questions, specifically from
+the answer boxes in the search results. This yields a rich space of answer types, containing both
+textual answers (short and long) as well as more structured ones such as collections.
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = ""
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+_URL = "https://github.com/allenai/gooaq/raw/main/data/qoogle.jsonl"
+
+
+class GooAQ(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class Gooaq(datasets.GeneratorBasedBuilder):
```
This must be the camel case version of the dataset name `gooaq`.
You might need to regenerate the dataset_infos.json file after this change